### PR TITLE
Databricks replace bug

### PR DESCRIPTION
**Pull request by djagoda881**
## Summary

Pr includes fixing the overwriting of tables in Databricks. Previously, a table was overwritten if it had the same number of columns. However, if, for example, a metadata column was added, overwriting was not possible. This has been fixed in this PR. I have also added a unit test for this case. It also seems to me that the tests need refactoring here, which is why I have created a user story for this [ user story ](https://dev.azure.com/VELUX-IT-Dev/AIA/_backlogs/backlog/AIA%20Team/Epics/?workitem=131621).

## Importance

This is important because any change to the table schema will cause an error if an attempt is made to overwrite it.

## Checklist

This PR:

- [ x ] adds new tests (if appropriate)
- [  ] updates docstrings for any new functions or function arguments (if appropriate)
- [ x ] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,10 +6,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 ### Added
+- Fixed `_full_trfresh` method in `Databricks` source.
```
Comment (Trymzet) [inline CHANGELOG.md:9]: 1) typo, 2) don't add internal methods to changelog

File: tests/unit/test_databricks.py [hunk 2]
```diff
@@ -209,6 +209,29 @@ def test_create_table_replace(databricks):
     )
     assert replaced is True
 
+    databricks.drop_table(schema=TEST_SCHEMA, table=TEST_TABLE)
+    databricks.drop_schema(TEST_SCHEMA)
+
+
+def test_replace_not_the_same_column_schema(databricks):
```
Comment (Trymzet) [inline tests/unit/test_databricks.py:216]: I would replace "not_the_same" with "different"

File: tests/unit/test_databricks.py [hunk 3]
```diff
@@ -209,6 +209,29 @@ def test_create_table_replace(databricks):
...
+
+    assert not databricks._check_if_table_exists(schema=TEST_SCHEMA, table=TEST_TABLE)
+
+    databricks.create_schema(TEST_SCHEMA)
+    databricks.create_table_from_pandas(
+        schema=TEST_SCHEMA, table=TEST_TABLE, df=TEST_DF
+    )
+
+    # Adds one additional column so that the table to be overwritten differs from our new table in the number of columns.
```
Comment (Trymzet) [inline tests/unit/test_databricks.py:225]: This line seems long?

**Review by Trymzet — CHANGES_REQUESTED** [review]
Very minor changes requested; the logic seems good.

File: tests/unit/test_databricks.py (hunk 2 above)
Comment (djagoda881) [reply tests/unit/test_databricks.py]: [✅ Updated name and comment i databricks tests](https://github.com/dyvenia/viadot/pull/658/commits/dc279bf072230139794efe2ac3719ed64f972316)

File: tests/unit/test_databricks.py (hunk 3 above)
Comment (djagoda881) [reply tests/unit/test_databricks.py]: [✅ Updated name and comment i databricks tests](https://github.com/dyvenia/viadot/pull/658/commits/dc279bf072230139794efe2ac3719ed64f972316)

File: CHANGELOG.md (hunk 1 above)
Comment (Trymzet) [reply CHANGELOG.md]: yes but still add the change :D I guess sth like "Fixed `Databricks.create_table_from_pandas()` failing to overwrite a table in some cases even with `replace="True"`"?

**Review by Trymzet — CHANGES_REQUESTED** [review]
Just the changelog entry missing

File: CHANGELOG.md (hunk 1 above)
Comment (Trymzet) [reply CHANGELOG.md]: Also you can always describe the change briefly and link the issue for more details, like `Fixed an issue with overriding tables in `Databricks.create_table_from_pandas()`, see  #issue_no`

Comment (djagoda881) [reply CHANGELOG.md]: [📝 Added entry to CHANGELOG.md](https://github.com/dyvenia/viadot/pull/658/commits/0a6cac43e19dc64421a40a6462a7af5cd046b69f)

**Review by Trymzet — APPROVED** [review]
Looks good

**Pull request merged**